src/main/java/com/synopsys/integration/alert/web/api/audit/AuditEntryActions.java

```diff
@@ -31,102 +31,135 @@
-        AlertNotificationModel notificationContent = notificationManager
-                                                         .findById(notificationId)
-                                                         .orElseThrow(() -> new AlertNotificationPurgedException("No notification with this id exists."));
-        List<DistributionEvent> distributionEvents;
-        if (null != commonConfigId) {
-            ConfigurationJobModel commonDistributionConfig = jobConfigReader.getJobById(commonConfigId).orElseThrow(() -> {
-                logger.warn("The Distribution Job with Id {} could not be found. This notification could not be sent", commonConfigId);
-                return new AlertJobMissingException("The Distribution Job with this id could not be found.", commonConfigId);
-            });
-            if (commonDistributionConfig.isEnabled()) {
-                distributionEvents = notificationProcessor.processNotifications(commonDistributionConfig, List.of(notificationContent));
+    public ActionResponse<AuditEntryPageModel> resendNotification(Long notificationId, UUID commonConfigId) {
+        if (!authorizationManager.hasExecutePermission(ConfigContextEnum.GLOBAL.name(), descriptorKey.getUniversalKey())) {
+            return new ActionResponse<>(HttpStatus.FORBIDDEN, ActionResponse.FORBIDDEN_MESSAGE);
+        }
+        try {
+            Optional<AlertNotificationModel> notification = notificationManager
+                                                                .findById(notificationId);
+            if (notification.isEmpty()) {
+                return new ActionResponse<>(HttpStatus.GONE, "No notification with this id exists.");
+            }
+            AlertNotificationModel notificationContent = notification.get();
+
+            List<DistributionEvent> distributionEvents;
+            if (null != commonConfigId) {
+                Optional<ConfigurationJobModel> commonDistributionConfig = jobConfigReader.getJobById(commonConfigId);
+                if (commonDistributionConfig.isEmpty()) {
+                    String message = String.format("The Distribution Job with this id could not be found. %s", commonConfigId.toString());
+                    return new ActionResponse<>(HttpStatus.GONE, message);
+                }
+                ConfigurationJobModel commonConfig = commonDistributionConfig.get();
+                if (commonConfig.isEnabled()) {
+                    distributionEvents = notificationProcessor.processNotifications(commonConfig, List.of(notificationContent));
+                } else {
+                    UUID jobConfigId = commonConfig.getJobId();
+                    logger.warn("The Distribution Job with Id {} was disabled. This notification could not be sent", jobConfigId);
+                    String message = String.format("The Distribution Job is currently disabled. %s", jobConfigId.toString());
+                    return new ActionResponse<>(HttpStatus.GONE, message);
```

Comment:
If the job is disabled, shouldn't this be a BAD_REQUEST response instead of GONE?

Reply:
Given that the job is disabled and not deleted then BAD_REQUEST makes sense rather than GONE.